matchbox_socket/src/webrtc_socket/mod.rs

```diff
@@ -160,26 +161,32 @@ async fn message_loop<M: Messenger>(
 
         select! {
             _  = &mut timeout => {
-                requests_sender.unbounded_send(PeerRequest::KeepAlive).expect("send failed");
-                // UNWRAP: we will only ever get here if there already was a timeout
-                let interval = keep_alive_interval.unwrap();
-                timeout = Either::Left(Delay::new(interval)).fuse();
+                requests_sender.unbounded_send(PeerRequest::KeepAlive).map_err(TrySendError::into_send_error)?;
+                if let Some(interval) = keep_alive_interval {
+                    timeout = Either::Left(Delay::new(interval)).fuse();
+                }
             }
 
             message = events_receiver.next().fuse() => {
                 if let Some(event) = message {
                     debug!("{event:?}");
                     match event {
                         PeerEvent::IdAssigned(peer_uuid) => {
-                            id_tx.try_send(peer_uuid.to_owned()).unwrap();
+                            if id_tx.take().expect("already sent peer id").send(peer_uuid.to_owned()).is_err() {
+                                // Socket receiver was dropped, exit cleanly.
+                                break Ok(());
+                            };
                         },
                         PeerEvent::NewPeer(peer_uuid) => {
                             let (signal_tx, signal_rx) = futures_channel::mpsc::unbounded();
                             handshake_signals.insert(peer_uuid, signal_tx);
                             let signal_peer = SignalPeer::new(peer_uuid, requests_sender.clone());
                             handshakes.push(M::offer_handshake(signal_peer, signal_rx, messages_from_peers_tx.clone(), ice_server_config, channel_configs))
                         },
-                        PeerEvent::PeerLeft(peer_uuid) => {peer_state_tx.unbounded_send((peer_uuid, PeerState::Disconnected)).expect("fail to report peer as disconnected");},
+                        PeerEvent::PeerLeft(peer_uuid) => {
+                            peer_state_tx.unbounded_send((peer_uuid, PeerState::Disconnected))
+                            .map_err(TrySendError::into_send_error)?;
```

Comment:
Is this the same case, that it can only fail if the socket is dropped?

Reply:
Yes, so propagating the error up seems reasonable.

Reply:
Also, I'm still not really convinced and `Err` is appropriate in this case (see example). What's your take on this @garryod ?